File: frontend/src/components/lib/friends.api.ts
// frontend/src/components/lib/friends.api.ts
import { api } from "./api";

export type FriendDTO = {
  _id: string;
  email?: string;
  firstName?: string;
  lastName?: string;
  // add fields as needed based on backend response
};

export type FriendsResponse = {
  friends: FriendDTO[];
  sentRequests: FriendDTO[];
  receivedRequests: FriendDTO[];
};

export async function getFriendsReal(): Promise<FriendsResponse> {
  const token = localStorage.getItem("token_data");
  // backend returns: { friends: [...] }
  const res = await api("/api/getfriends", {
    method: "GET",
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

  return res.json();
}

export async function acceptFriendReal(requesterId: string) {
  const token = localStorage.getItem("token_data");

  const res = await api("/api/acceptfriend", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`,
    },
    body: JSON.stringify({ requesterId }),
  });

  return res.json();
}

export async function declineFriendReal(requesterId: string) {
  const token = localStorage.getItem("token_data");

  const res = await api("/api/declinefriend", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`,
    },
    body: JSON.stringify({ requesterId }),
  });

  return res.json();
}


export async function addFriendReal(friendEmail: string): Promise<{ message: string }> {
  const res = await api("/api/addfriend", {
    method: "POST",
    body: JSON.stringify({ friendEmail }),
  });
  return res.json();
}

export async function removeFriendReal(friendEmail: string): Promise<{ message: string }> {
  const res = await api("/api/removefriend", {
    method: "POST",
    body: JSON.stringify({ friendEmail }),
  });
  return res.json();
}

File: mobile/lib/services/friends_api.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:mobile/services/api_service.dart';

class FriendsApi {
  /// ---------------------------
  /// GET FRIENDS LIST
  /// ---------------------------
  static Future<Map<String, dynamic>> getFriends() async {
    final res = await ApiService.api('/getfriends');

    if (res.statusCode == 200) {
      return json.decode(res.body);
    } else {
      throw Exception('Failed to load friends');
    }
  }

  /// ---------------------------
  /// ADD FRIEND (by email)
  /// ---------------------------
  static Future<Map<String, dynamic>> addFriend(String friendEmail) async {
    final res = await ApiService.api(
      '/addfriend',
      method: 'POST',
      body: {'friendEmail': friendEmail},
    );

    return json.decode(res.body);
  }

  /// ---------------------------
  /// ACCEPT FRIEND REQUEST
  /// ---------------------------
  static Future<Map<String, dynamic>> acceptFriend(String requesterId) async {
    final res = await ApiService.api(
      '/acceptfriend',
      method: 'POST',
      body: {'requesterId': requesterId},
    );

    return json.decode(res.body);
  }

  /// ---------------------------
  /// DECLINE FRIEND REQUEST
  /// ---------------------------
  static Future<Map<String, dynamic>> declineFriend(String requesterId) async {
    final res = await ApiService.api(
      '/declinefriend',
      method: 'POST',
      body: {'requesterId': requesterId},
    );

    return json.decode(res.body);
  }

  /// ---------------------------
  /// REMOVE FRIEND
  /// ---------------------------
  static Future<Map<String, dynamic>> removeFriend(String friendEmail) async {
    final res = await ApiService.api(
      '/removefriend',
      method: 'POST',
      body: {'friendEmail': friendEmail},
    );

    return json.decode(res.body);
  }
}
